src/utils/gettopkclasses.js

```diff
@@ -0,0 +1,38 @@
+// Copyright (c) 2018 ml5
+//
+// This software is released under the MIT License.
+// https://opensource.org/licenses/MIT
+
+export function getTopKClassesFromArray(values, topK, CLASSES) {
+  const valuesAndIndices = [];
+  for (let i = 0; i < values.length; i += 1) {
+    valuesAndIndices.push({
+      value: values[i],
+      index: i,
+    });
+  }
+  valuesAndIndices.sort((a, b) => b.value - a.value);
+
+  const topkValues = new Float32Array(topK);
+  const topkIndices = new Int32Array(topK);
+  for (let i = 0; i < topK; i += 1) {
+    topkValues[i] = valuesAndIndices[i].value;
+    topkIndices[i] = valuesAndIndices[i].index;
+  }
+
+  const topClassesAndProbs = [];
+  for (let i = 0; i < topkIndices.length; i += 1) {
+    topClassesAndProbs.push({
+      className: CLASSES[topkIndices[i]],
+      probability: topkValues[i],
+    });
+  }
+  return topClassesAndProbs;
+}
+
+export async function getTopKClassesFromTensor(logits, topK, CLASSES) {
+  const values = await logits.data();
+  return getTopKClassesFromArray(values, topK, CLASSES);
+}
+
+export default { getTopKClassesFromArray, getTopKClassesFromTensor }
```

Comment:
It is so frustrating to see all the work that I have put into this library being completely ignored.  I rewrote this utility to be cleaner, more efficient, better documented, and handle more situations.  Why are we still using the old version?

https://github.com/ml5js/ml5-library/blob/a0ba76f5f11a1f39e5311985b55a4edb4f14d22b/src/utils/gettopkclasses.js

https://github.com/ml5js/ml5-library/pull/1364